diff --git a/src/node-haste/DependencyGraph.js b/src/node-haste/DependencyGraph.js
index 8222d8c69059bfb3fe444acba9f96ab2cc7bec0b..489d5f0fff5877f5ecdb14a9b7e3bf8afb999a51 100644
--- a/src/node-haste/DependencyGraph.js
+++ b/src/node-haste/DependencyGraph.js
@@ -32,6 +32,11 @@ function getOrCreateMap(map, field) {
   }
   return subMap;
 }
+
+const workletsDirPath = _path.default.join(
+  "react-native-worklets",
+  ".worklets",
+);
 class DependencyGraph extends _events.default {
   #packageCache;
   #dependencyPlugin;
@@ -197,6 +202,18 @@ class DependencyGraph extends _events.default {
     return (0, _nullthrows.default)(this._fileSystem).getAllFiles();
   }
   async getOrComputeSha1(mixedPath) {
+    // Bundle Mode isolates each worklet in a separate module (file) created on
+    // the fly, which Metro doesn't index in time. Short-circuit with a fresh
+    // hash so the modules always (re)transform instead of failing the build.
+    // Official workaround: https://docs.swmansion.com/react-native-worklets/docs/bundleMode/setup
+    if (mixedPath.includes(workletsDirPath)) {
+      const createHash = require("crypto").createHash;
+      return {
+        sha1: createHash("sha1")
+          .update(performance.now().toString())
+          .digest("hex"),
+      };
+    }
     const result = await this._fileSystem.getOrComputeSha1(mixedPath);
     if (!result || !result.sha1) {
       throw new Error(`Failed to get the SHA-1 for: ${mixedPath}.
